src/anomalib/engine/engine.py

```diff
@@ -350,6 +350,10 @@ def test(
         if model:
             self._setup_trainer(model)
             self._setup_dataset_task(dataloaders)
+        if ckpt_path is None:
```

Comment:
What if someone calls the API like this
```python
model = Padim()
engine.fit(model, train_dataloader)
engine.test(model, test_dataloader)
```
Maybe we need to already consider the "model type" in the design. Or maybe extend `is_fitted` to all the models